src/main/java/core/basesyntax/Figure.java

```diff
@@ -0,0 +1,26 @@
+package core.basesyntax;
+
+public abstract class Figure implements Drawable {
+    private Color color;
+
+    public Figure(Color color) {
+        this.color = color;
+    }
+
+    public Color getColor() {
+        return color;
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure: " + getShape().getDescription().toLowerCase()
+                + ", area: " + String.format("%.2f", getArea()) + " sq.units, "
+                + getSizeDescription() + ", color: " + getColor().name().toLowerCase());
+    }
+
+    public abstract double getArea();
+
+    public abstract Shape getShape();
+
+    public abstract String getSizeDescription();
```

Comment:
This method is just used to extract part of text from draw() method.
This method doesn't provide any functionality that is aimed to be used with public access.
Also it's not good that we're forcing to follow our code-style pattern in such way. Other programmers may want to use their own code-style, probably without extracting text from draw() method, when creating new child class for our Figure. 
Please just use private methods in child classes if you really want to extract something from public method

Reply:
Agree, these methods may not be public if we consider their usage.
Protected may look better.

getShape() and getSizeDescription() are aimed to prevent duplicating code of draw() in every Figure child class.

If we follow the idea of not forcing other programmers to write these methods and ask them to write their own version of the draw() method, we could place these methods into a dedicated interface. 
Please take a look at fielddescriptor branch. 
Do you think that version is better? 
In my opinion, it looks awkward and overdesigned for this task purpose.